test/validation/api/crypto/odp_crypto_test_inp.c

```diff
@@ -977,10 +1035,7 @@ static odp_crypto_session_t session_create(odp_crypto_op_t op,
 	}
 
 	/*
-	 * We do not allow ODP_CRYPTO_SES_ERR_ALG_ORDER since we do
-	 * not combine individual non-null crypto and auth algorithms
-	 * with each other in the tests. Both orders should work when
-	 * only one algorithm is used (i.e. the other one is null).
+	 * We do not allow ODP_CRYPTO_SES_ERR_ALG_ORDER.
```

Comment:
I think this may also need some adjustments.

Earlier, the UT would attempt only single algo, ie, any non-NULL algo + NULL. So, the order would not matter. But now that we attempt chained cases, we should handle ALG_ORDER also the same way as ALG_COMBO.

Reply:
I think an ODP implementation should be able to support both orders by resorting internally to two individual crypto operations. But since that may not be trivial in the async mode, I changed the code so that ALG_ORDER error is allowed without test failure in async operation mode.